src/org/opensolaris/opengrok/web/Statistics.java

```diff
@@ -103,11 +126,11 @@ protected void maybeRefresh() {
      * @param category category
      */
     public void addRequest(HttpServletRequest req, String category) {
-        Long val = requestCategories.get(category);
+        AtomicLong val = requestCategories.get(category);
         if (val == null) {
-            val = new Long(0);
+            val = new AtomicLong(0);
```

Comment:
The design with the synchrinized blocks/methods was simpler in my opinion than this